src/Navigation/WPF/Navigation.WPF/Service/NavigationService.registration.cs

```diff
@@ -131,47 +164,59 @@ private static void RegisterInternal(INavigationRegistration registration)
 
             registrationInfo = NavigationRegistrations.FirstOrDefault(registration =>
                 registration.ViewModelType == viewModelType &&
-                viewType.IsAssignableFrom(registration.ViewType));
+                viewType.IsAssignableFrom(registration.ViewType) &&
+                registration.Condition?.Invoke() != false);
 
             return registrationInfo != null;
         }
 
-        protected void ThrowIfNotRegistered<TViewModelInterface>(
-            Type typeofView = null)
+
+        private static void ThrowIfNotRegistered<TViewModel>()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'ThrowIfNotRegistered'.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)